test/integration/api/applications_controller_test.rb

```diff
@@ -4,123 +4,155 @@
 
 class Api::ApplicationsControllerTest < ActionDispatch::IntegrationTest
 
-  setup do
-    @provider = FactoryGirl.create(:provider_account)
-    login! @provider
-  end
-
-  attr_reader :provider
-
-  class Index < Api::ApplicationsControllerTest
+  class MasterLoggedInTest < Api::ApplicationsControllerTest
     setup do
-      @service = provider.services.first!
-      plans = FactoryGirl.create_list(:application_plan, 2, service: @service)
-      buyers = FactoryGirl.create_list(:buyer_account, 2, provider_account: provider)
-      plans.each_with_index { |plan, index| buyers[index].buy! plan }
+      login! master_account
+      @service = master_account.default_service
+      FactoryGirl.create(:cinstance, service: @service)
     end
 
     attr_reader :service
 
-    test 'index retrieves all cinstances of a service' do
+    test 'index retrieves all master\'s provided cinstances except those whose buyer is master' do
       get admin_service_applications_path(service)
 
       assert_response :ok
-      assert_same_elements service.cinstances.pluck(:id), assigns(:cinstances).map(&:id)
-    end
-
-    test 'index can retrieve the cinstances of an application plan belonging to a service' do
-      plan = service.application_plans.first!
-      get admin_service_applications_path(service, {application_plan_id: plan.id})
-
-      assert_response :ok
-      assert_same_elements plan.cinstances.pluck(:id), assigns(:cinstances).map(&:id)
-    end
-
-    test 'index can retrieve the cinstances of a buyer account' do
-      buyer = @provider.buyers.last!
-      get admin_service_applications_path(service, {account_id: buyer.id})
-
-      assert_response :ok
-      assert_same_elements buyer.bought_cinstances.pluck(:id), assigns(:cinstances).map(&:id)
+      expected_cinstances_ids = service.cinstances.where.has { user_account_id != Account.master.id }.pluck(:id)
```

Comment:
Isn't there a way to explicitly check who's provider and who's buyer? Like `service.cinstance.where.is { provider }`

Reply:
we spoke about it face to face and the conclusion is it is not clear that `accessible_cinstances` means `provided_cinstances` but with restrictions, but that doesn't belong to this PR, however, it is not clear that the `where.has { user_account_id != Account.master.id }.pluck(:id)` means where it is not its own buyer, so i will make that clearer 👍

Reply:
Done 👍 But I made it clearer in the code itself (in https://github.com/3scale/porta/pull/307/files#diff-92c5f23a70067c2e040ae0a97630516cR54), in the test it is correct imo, because the account logged in is master account in this test 😄

Reply:
Probably `not_bought_by_master`?

Reply:
That's not what it is doing, it is checking that it is not bought by the same account as it is logged in, no matter if it is master or another one, and IMO that makes more sense for what we want 🤔